### PR TITLE
ipa_tests: test subca key replication

### PR DESCRIPTION
**Pull request by Rezney**
Test if key replication is not failing.

https://pagure.io/freeipa/issue/7387

### CODE REVIEW

**Comment by Rezney** [comment]
Wondering whether we should create a new e.g. "test_certificates.py" module in order to not "pollute" "test_replica_promotion.py" with other sort of tests. Any comments?

File: ipatests/test_integration/test_replica_promotion.py [hunk 1]
```diff
@@ -570,3 +570,50 @@ def test_replica_install_with_existing_entry(self):
...
+
+    def test_sub_ca_key_replication(self):
+        master = self.master
+        replica = self.replicas[0]
+
+        result = master.run_command(['ipa', 'ca-add', self.SUBCA, '--subject',
+                                     self.SUBCA_CN])
+        assert result.returncode == 0
+
+        uuid = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
```
Comment (tiran) [inline ipatests/test_integration/test_replica_promotion.py]: Traditionally, regular expression constants and compiled re objects are suffixed with ``_re``, e.g. ``auth_id_re``. It's a bit confusing to have two variables with same name ``auth_id`` but different meaning.

File: ipatests/test_integration/test_replica_promotion.py [hunk 2]
```diff
@@ -570,3 +570,50 @@ def test_replica_install_with_existing_entry(self):
...
+        auth_id = re.compile('Authority ID: ({})'.format(uuid), re.IGNORECASE)
+        auth_id = "".join(re.findall(auth_id, result.stdout_text))
+
+        cert_nick = '{} {}'.format(IPA_CA_NICKNAME, auth_id)
```
Comment (tiran) [inline ipatests/test_integration/test_replica_promotion.py]: Bonus points for new format operator 👍

File: ipatests/test_integration/test_replica_promotion.py [hunk 3]
```diff
@@ -570,3 +570,50 @@ def test_replica_install_with_existing_entry(self):
...
+
+        # give replication some time
+        time.sleep(30)
+
+        result = replica.run_command(['ipa-certupdate'])
+        assert result.returncode == 0
```
Comment (tiran) [inline ipatests/test_integration/test_replica_promotion.py]: You don't need to clutter the code with ``assert result.returncode == 0``. pytest multihost's ``run_command`` method defaults to ``raiseonerr=True``, https://pagure.io/python-pytest-multihost/blob/master/f/pytest_multihost/host.py#_202

File: ipatests/test_integration/test_replica_promotion.py (hunk 1 above)
Comment (Rezney) [reply ipatests/test_integration/test_replica_promotion.py]: yep, makes sense...

File: ipatests/test_integration/test_replica_promotion.py (hunk 3 above)
Comment (Rezney) [reply ipatests/test_integration/test_replica_promotion.py]: keep forgetting about this or just like "asserts" :)

**Comment by tiran** [comment]
master:

* e5a508a749364f6a8916676061185c62972ea4a9 ipa_tests: test subca key replication

**Pull request closed without merging**